frontend/src/pages/CompareV2.tsx

```diff
@@ -233,9 +239,65 @@ function VisualizationPlaceholder(props: VisualizationPlaceholderProps) {
   );
 }
 
+interface VisualizationPanelItemProps {
+  metricsTab: MetricsType;
+  metricsTabText: string;
+  linkedArtifact: LinkedArtifact | undefined;
+  namespace: string | undefined;
+}
+
+function VisualizationPanelItem(props: VisualizationPanelItemProps) {
+  const { metricsTab, metricsTabText, linkedArtifact, namespace } = props;
+
+  const { data: viewerConfigs } = useQuery<ViewerConfig[], Error>(
+    [
+      'viewerConfig',
+      {
+        artifact: linkedArtifact?.artifact.getId(),
+        namespace,
+      },
+    ],
+    async () => {
+      let viewerConfigs: ViewerConfig[] = [];
+      if (linkedArtifact) {
+        try {
+          if (metricsTab === MetricsType.HTML) {
+            viewerConfigs = await getHtmlViewerConfig([linkedArtifact], namespace);
+          } else if (metricsTab === MetricsType.MARKDOWN) {
+            viewerConfigs = await getMarkdownViewerConfig([linkedArtifact], namespace);
+          }
+        } catch (err) {
+          throw err;
+        }
+      }
+      return viewerConfigs;
+    },
+    { staleTime: Infinity },
+  );
+
+  if (!linkedArtifact) {
+    return <VisualizationPlaceholder metricsTabText={metricsTabText} />;
+  }
+
+  if (metricsTab === MetricsType.CONFUSION_MATRIX) {
+    return (
+      <React.Fragment key={linkedArtifact.artifact.getId()}>
```

Comment:
Question: Do we need this Fragment? Is it needed to have `key`?

Reply:
Yes it is - I originally included it based again on the `MetricsVisualizations.tsx` reference: https://github.com/kubeflow/pipelines/blob/49c3587591fc291dc4b0f85ea7339d330ffe451c/frontend/src/components/viewers/MetricsVisualizations.tsx#L174-L182

On further examination, this `React.Fragment` forces the confusion matrix to re-render when a new artifact is selected (but does not re-render when the same artifact is re-selected, since the artifact ID did not change.

When `React.Fragment` is not included, the confusion matrix does not re-render upon new select.

Reply:
There may be a way to prompt a re-render without `React.Fragment` by adding a `useEffect` to the `ConfusionMatrixSection` component, but I think this solution may be longer than the current code - I am flexible though, this is essentially what I added for the HTML and Markdown files with (`componentDidUpdate`).

Reply:
Thank you for the examination! I think it is a reasonable enough to enable re-rendering using fragment. Let's not make changes here.